src/components/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,223 @@
+                                               alternative=false,
+                                               verticalStepper=false,
+                                               stepsArr,
+                                               detectError,
+                                               detectOptional,
+                                               customErrorSvg,
+                                               customFinishedSvg,
+                                               children}) => {
+
+    const [activeStep, setActiveStep] = useState<number>(0);
+    const [skipped, setSkipped] = useState(new Set<number>());
+    const [errorSteps, setErrorSteps] = useState(new Set<number>());
+
+
+    const checkIsOptional = (step: number): boolean => {
+        if (detectOptional) {
+            return detectOptional(step);
+        } else {
+            return false;
+        }
+    }
+
+    const checkIsError = (step: number) => {
+        if (detectError) {
+            return detectError(step);
+        } else {
+            return false;
+        }
+    }
+
+    const checkIsSKipped = (step: number) => {
+        return skipped.has(step);
+    }
+
+    const handleSkipClick = () => {
+        if (!checkIsOptional(activeStep)) {
+            throw new Error('You can\'t skip a step that is not optional.')
+        }
+        setActiveStep(prevState => prevState + 1);
+        setSkipped(prevState => {
+            const newSkipped = new Set(prevState.values())
+            newSkipped.add(activeStep);
+            return newSkipped;
+        })
+    }
+
+    const handleBackClick = () => {
+        setActiveStep(prevState => prevState - 1);
+    }
+
+    const handleNextClick = (step: number) => {
+        const isError = checkIsError(step);
+        if (!isError) {
+            setActiveStep(prevState => prevState + 1);
+        } else {
+            setErrorSteps(errorSteps.add(step));
+            forceReRender();
+        }
+    }
+
+    const handleResetClick = () => {
+        setSkipped(new Set());
+        setErrorSteps(new Set());
+        setActiveStep(0);
+    }
+
+
+    let verticalStyle = classNames('stepper-container--vertical');
+    let horizontalStyle = classNames('stepper-container--horizontal');
+    let horizontalItemStyle = classNames('stepper-item-container--horizontal');
+    let horizontalContentStyle = classNames('stepper-content-container--horizontal');
+
+    verticalStyle = verticalContainerClassName ? verticalStyle + ' ' + verticalContainerClassName : verticalStyle;
+    horizontalStyle = horizontalContainerClassName ? horizontalStyle + ' ' + horizontalContainerClassName : horizontalStyle;
+    horizontalItemStyle = horizontalItemClassName ? horizontalItemStyle + ' ' + horizontalItemClassName : horizontalItemStyle;
+    horizontalContentStyle = horizontalContentClassName ? horizontalContentStyle + ' ' + horizontalContentClassName : horizontalContentStyle;
+
+    return verticalStepper ? (
```

Comment:
I think for rendering a vertical stepper, changing the stepper style by adding scss classes and overriding certain styles like flex-direction instead of writing a different component would make the code more maintainable.